R/utility_spliceHelper_functions.R

```diff
@@ -1,598 +1,572 @@
+  subjectEndRng <- selectEndExonFromRangesList(subjectFullRng, strand)
+  querySpliceRng <- ranges(myGaps(query))
+  querySpliceRng[elementNROWS(querySpliceRng)==0] <- IRanges(start=1,end=1) # add mock intron
+  subjectSpliceRng <- ranges(myGaps(subject))
+  subjectSpliceRng[elementNROWS(subjectSpliceRng)==0] <- IRanges(start=1,end=1)# add mock intron
+  annotatedTable <- tibble(queryId = names(query), subjectId = names(subject), strand = strand)
+  # calculate alternative First/last exons and annotate internal start and end first exons
+  annotatedTable$alternativeFirstExon <- alternativeStartEndExon(queryStartRng, 
+                                                                subjectStartRng)
+  annotatedTable$alternativeTSS <- calculateTSSdistance(queryStartRng, 
+                                                       subjectStartRng, 
+                                                       annotatedTable$alternativeFirstExon,
+                                                       strand)
+  annotatedTable$internalFirstExon.query <- annotateInternalStartEnd(queryStartRng,
+                                                                    subjectFullRng, annotatedTable$alternativeFirstExon)
+  annotatedTable$internalFirstExon.subject <- annotateInternalStartEnd(subjectStartRng,
+                                                                      queryFullRng, annotatedTable$alternativeFirstExon)
+  # TES/last exon
+  annotatedTable$alternativeLastExon <- alternativeStartEndExon(queryEndRng, 
+                                                               subjectEndRng)
+  annotatedTable$alternativeTES <- calculateTESdistance(queryEndRng, 
+                                                       subjectEndRng, 
+                                                       annotatedTable$alternativeLastExon,
+                                                       strand)
+  annotatedTable$internalLastExon.query <- annotateInternalStartEnd(queryEndRng,
+                                                                   subjectFullRng, annotatedTable$alternativeLastExon)
+  annotatedTable$internalLastExon.subject <- annotateInternalStartEnd(subjectEndRng,
+                                                                     queryFullRng, annotatedTable$alternativeLastExon)
+  #intron retention subject and query
+  annotatedTable$intronRetention.subject <- annotateIntronRetent(querySpliceRng, subjectFullRng)
+  annotatedTable$intronRetention.query <- annotateIntronRetent(subjectSpliceRng ,queryFullRng)
+  #exon skipping query and subject
+  annotatedTable$exonSkipping.query <- annotateExonSkip(querySpliceRng, subjectFullRng,
+                                                        subjectStartRng, subjectEndRng)
+  annotatedTable$exonSkipping.subject <- annotateExonSkip(subjectSpliceRng ,queryFullRng,
+                                                          queryStartRng, queryEndRng) 
+  # exon 3' splice site
```

Comment:
what about excon 5' splice site? Will both be done in this function? Then that might be good to include in the comment

Reply:
I have updated the comment to include 5' splice sites